chef/chef-server-data-generator
# Basic benchmarking of solr indexing on nodes.

To test out the benchmarking I have so far:

```
ssh into running chef server
git clone this
git checkout tc/benchmark
cp /etc/opscode/pivotal.rb .chef/knife-in-guest.rb
mv setup.yml.example setup.yml and change any ec settings
./setup-ec.sh (generates data on the chef-server)
time ./benchmark.sh
```

Currently its only benchmarking updating node attributes. Look in `testdata/created-objects.yml` after `./setup-ec.sh` to see what data is generated.

To add benchmarking, check out `scripts/ec/benchmark/solr.rb` and `benchmark.sh`, but basically, the latter is iterating through the admin user knife files that are generated (there is admin user / knife file per org generated).

If you wanna start from scratch with fresh data, `rm -rf testdata/*` and re-run `./setup-ec.sh` and `time ./benchmark.sh` (we should make this better).
